assets/js/modules/analytics-4/datastore/properties.js

```diff
@@ -99,6 +101,38 @@ const baseActions = {
 			return { response, error };
 		}() );
 	},
+
+	/**
+	 * Sets the given property and related fields in the store.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} propertyID GA4 property ID.
+	 * @return {Object} A Generator function.
+	 */
+	selectProperty( propertyID ) {
+		invariant( isValidPropertySelection( propertyID ), 'A valid propertyID selection is required.' );
+
+		return ( function* () {
+			const registry = yield Data.commonActions.getRegistry();
+
+			registry.dispatch( STORE_NAME ).setPropertyID( propertyID );
+			registry.dispatch( STORE_NAME ).setWebDataStreamID( '' );
+			registry.dispatch( STORE_NAME ).setMeasurementID( '' );
```

Comment:
It's important that we always set these individually rather than using `setSettings` since this updates the object with _all settings_, it doesn't merge them. Using `setSettings` here would unset others like `useSnippet` or any other new setting in the future, so it's important that we always update them individually whenever we're not working with the full settings object.